### DISCUSSION TITLE
How I can rename my input file using data from input message after processed

### DISCUSSION THREAD

**Discussion by IbrahimAlhaj9** [nextgenhealthcare/connect#5100] [General]
Hi Everyone, I need to know how i can rename my input file using data from the input hl7 message. for example I want to set the name of the file based on the value of MSH.7.1 and MSH.10.1.
![123](https://user-images.githubusercontent.com/54806435/158551095-4fc70fd7-43d5-4bc7-a2f0-cdab51ddb3fb.PNG)

please help.
Thank you.

**Comment by pacmano1** [comment]
If you are creating a channel map variable and expect to use it there, you can't.   

If you really need to rename it to something you get/set to a channel map variable as part of the message, move the file after processing with java.  I think https://forums.mirthproject.io/forum/mirth-connect/support/6756-file-reader-move?p=47772#post47772 describes something similar to your use case.

**Comment by IbrahimAlhaj9** [comment]
@pacmano1 thank you for your feedback.
I am trying the solution that mentioned in the link but I gave the following error
```
DETAILS:	Can't find method org.apache.commons.io.FileUtils.moveFile(string,java.io.File).
```

the problem is from :FileUtils.moveFile(src,dest);

How I can import the missing package ? I need the following one :
import org.apache.commons.io.FileUtils;

Thank YOU

**Comment by pacmano1** [comment]
in 3.12, you can

```JS
importPackage(org.apache.commons.io);

FileUtils.moveFile(FileUtils.getFile('/tmp/text.txt'), FileUtils.getFile('/tmp/test1.txt'))
```

**Comment by IbrahimAlhaj9** [comment]
@pacmano1 thanks it's working now when the source connector file is File Reader.
Can I do a similar change when the source connector is HTTP Listener? if yes, How?

Thank you

**Reply by pacmano1** [reply]
I don't understand your question.  HTTP listeners receive data that you then parse / consume.   So perhaps explain a bit more?

**Reply by IbrahimAlhaj9** [reply]
@pacmano1 In other words, can the mirth rename the input file name when using HTTP Listener?

**Reply by pacmano1** [reply]
There is no input file name for a http listener.   An http listener receives "stuff", like JSON, XML, raw data, etc. in headers, parameters and the data itself which may or not be encoded, multipart and so on.  You need to parse that to do something with it.   So your question does not make sense to me.

**Reply by IbrahimAlhaj9** [reply]
yes you are right
Thank you
